metrix/src/app.rs

```diff
@@ -108,13 +109,19 @@ fn aggregate_metrics_route(
         filter_clause
     );
     let results = sql_query(query_string)
-        .load(&db_conn)
+        .load::<Bucket>(&db_conn)
         .expect("Error loading metrics");
 
+    let mut padded_results: Vec<Bucket> = vec![();bucket_count as usize]
+        .iter().enumerate().map(|(i, bucket)| Bucket { value: 0, bucket: i as i32 }).collect();
+
+    for result in &results {
+        replace(&mut padded_results[result.bucket as usize], *result);
+    }
     return Ok(Json(
         BucketedData {
             data: Buckets {
-                buckets: results,
+                buckets: padded_results,
```

Comment:
The schema made me realize, the bucket shouldn't be the bucket index, that's implicit from the position in the array. The bucket should be the start of the time bucket. @woodgern

Reply:
Yeah I'm cool with that

Reply:
@kamikaz1k done